From 13fc758110e5dddd1d5810a4a1def2db9fd21e1a Mon Sep 17 00:00:00 2001
From: Ansuel Smith <ansuelsmth@gmail.com>
Date: Fri, 23 Jul 2021 15:09:38 +0200
Subject: [PATCH] NET: add prop mac-address from nvmem or mtd

Add "mac-address" prop with mtd or nvmem address source.

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
---
 drivers/of/of_net.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

--- a/drivers/of/of_net.c
+++ b/drivers/of/of_net.c
@@ -181,6 +181,7 @@ static int of_get_mac_address_mtd(struct
 */
 int of_get_mac_address(struct device_node *np, u8 *addr)
 {
+	struct property *prop;
 	u32 inc_idx, mac_inc;
 	int ret;
 
@@ -209,12 +210,24 @@ int of_get_mac_address(struct device_nod
 
 	ret = of_get_mac_address_mtd(np, addr);
 	if (!ret)
-		goto found;
+		goto set_prop;
 
 	ret = of_get_mac_addr_nvmem(np, addr);
 	if (ret)
 		return ret;
 
+set_prop:
+	prop = kzalloc(sizeof(*prop), GFP_KERNEL);
+	if (!prop)
+		return -ENOMEM;
+
+	prop->name = "mac-address";
+	prop->length = ETH_ALEN;
+	prop->value = addr;
+	if (of_add_property(np, prop)) {
+		kfree(prop);
+	}
+
 found:
 	if (!of_property_read_u32(np, "mac-address-increment", &mac_inc))
 		addr[inc_idx] += mac_inc;
